rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -430,26 +461,143 @@ def _decode_curve_gauge_events(self, context: DecoderContext) -> DecodingOutput:
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == transaction.from_address and
+                (
+                    event.address in CURVE_DEPOSIT_CONTRACTS or
+                    event.address == ZERO_ADDRESS
+                )
+            ):
+                receive_event_idx = idx
+                crypto_asset = event.asset.resolve_to_crypto_asset()
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_CURVE
+                event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} after depositing in curve pool'  # noqa: E501
+
+        if receive_event_idx is not None and len(deposit_events) > 0:
+            self._sort_events(
+                action_type='addition',
+                return_or_receive_event=decoded_events[receive_event_idx],
+                withdrawal_or_deposit_events=deposit_events,
+                all_events=decoded_events,
+            )
+            return
+
+        log.debug(
+            f'Expected to see a receive pool token event and deposit events for a curve pool, '
+            f'but have not found them. Tx_hash: {transaction.tx_hash.hex()}',
+        )
+
+    def _handle_zap_contracts_liquidity_removal(
+            self,
+            pool_address: ChecksumEvmAddress,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+    ) -> None:
+        """
+        This method completes the decoding of liquidity removal made through
+        Curve zap contracts such as e.g. Curve Deposit USDN, etc.
+        """
+        return_event_idx, withdrawal_events = None, []
+        for idx, event in enumerate(decoded_events):
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.RETURN_WRAPPED and
+                event.counterparty == CPT_CURVE and
+                event.extra_data is None
+            ):
+                return_event_idx = idx
+
+            elif (  # find receive events from the deposit contract.
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == transaction.from_address and
+                event.address in CURVE_DEPOSIT_CONTRACTS
+            ):
+                crypto_asset = event.asset.resolve_to_crypto_asset()
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_CURVE
+                event.notes = f'Remove {event.balance.amount} {crypto_asset.symbol} from the curve pool {pool_address}'  # noqa: E501
```

Comment:
Do it just like the rest of the file. Be consistent.

```suggestion
                event.notes = f'Remove {event.balance.amount} {crypto_asset.symbol} from {pool_address} curve pool'  # noqa: E501
```

Reply:
The rest of the file doesn't follow the format you suggested. Unless, you meant switching everything to what you suggested

Reply:
The gauges do. "The curve pool X" is not correct English. It's either:

1. from curve pool {name}
or
2. from {name} curve pool

And (2) is much more natural when you use it in a sentence.